### PR TITLE
Prune saved shifted instances

### PR DESCRIPTION
**Pull request by getzze**
Following #870, memory is continuously being filled up by saved shifted instances if `save_shifted_instances=True` (which is always the case if using Flow tracker from the GUI before #974 was merged).

This corrects this bug by pruning the dictionary of reference instances older than `track_window` before tracking a new frame.

Sorry for not seeing this bug before, if you are using a version between PR #870 and #974, you can have memory issues if you use the GUI and flow tracking.

### CODE REVIEW

**Review by roomrys — CHANGES_REQUESTED** [review]
Request to add a docstring, but otherwise looks good! Can we add a test for this section of code - I can help write this if you are ready to submit the PR, just let me know.

File: sleap/nn/tracking.py [hunk 1]
```diff
@@ -164,6 +169,19 @@ def get_candidates(
...
+            return
+        # Find ref_t older than track_window
+        keys_to_remove = []
+        for k in self.shifted_instances.keys():
+            if t - k[0] > self.track_window:
+                keys_to_remove.append(k)
+
+        # Delete items
+        for k in keys_to_remove:
+            del self.shifted_instances[k]
```
[the remark this answers is not in the document]
Comment (getzze) [reply sleap/nn/tracking.py]: Much better :)

File: sleap/nn/tracking.py [hunk 2]
```diff
@@ -105,6 +105,7 @@ class FlowCandidateMaker:
     of_window_size: int = 21
     of_max_levels: int = 3
     save_shifted_instances: bool = False
+    track_window: int = 5
```
[the remark this answers is not in the document]
Comment (getzze) [reply sleap/nn/tracking.py:125]: done

**Comment by getzze** [comment]
Thanks for reviewing the commit so fast!
I pushed the changes you proposed. 
Regarding the tests, I tried to write them some time ago, but I had a hard time figuring out how to create a mock video and all...

**Review by roomrys — APPROVED** [review]

**Pull request merged**